pkg/util/git.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"bytes"
+	"os/exec"
+
+	"istio.io/pkg/log"
+	"istio.io/release-builder/pkg/model"
+)
+
+// PushCommit will create a branch and push a commit with the specified commit text
+func PushCommit(manifest model.Manifest, repo, branch, commitString string, dryrun bool) (changes bool, err error) {
+	output := bytes.Buffer{}
+	cmd := VerboseCommand("git", "status", "--porcelain")
+	cmd.Dir = manifest.RepoDir(repo)
+	cmd.Stdout = &output
+	if err := cmd.Run(); err != nil {
+		return false, err
+	}
+	if output.Len() == 0 {
+		log.Infof("no changes found to commit")
+		return false, nil
+	}
+	log.Infof("changes found:\n%s", &output)
+
+	if !dryrun {
+		cmd = VerboseCommand("git", "checkout", "-b", branch)
```

Comment:
nit, but should creating a branch and pushing the commit be separate? Wondering if we might want to push multiple commits on the same branch.

Reply:
Certainly possible. The current flow is a single commit per step. This could be changed to a commit per substep, of which STEP 4 is the only step with substeps. In that case there would be a commit updating the common-files, another for updating the CODEOWNERS, etc.